Assets/MixedRealityToolkit.Tests/PlayModeTests/SolverTests.cs

```diff
@@ -0,0 +1,246 @@
+            testObjects.handler.TrackedTargetType = TrackedObjectType.Head;
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+
+            Assert.LessOrEqual(Vector3.Distance(testObjects.target.transform.position, Vector3.zero), DistanceThreshold);
+
+            // Test orbital around custom override
+            testObjects.handler.TrackedTargetType = TrackedObjectType.CustomOverride;
+            testObjects.handler.SetTransformOverride(transformOverride.transform);
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
+
+            Assert.LessOrEqual(Vector3.Distance(testObjects.target.transform.position, customTransformPos), DistanceThreshold);
+
+            yield return null;
+        }
+
+        /// <summary>
+        /// Tests solver handler's ability to switch hands
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestHandModality()
+        {
+            var testObjects = InstantiateTestSolver<Orbital>();
+
+            // Set solver handler to track hands
+            testObjects.handler.TrackedTargetType = TrackedObjectType.HandJoint;
+
+            // Set and save revelant positions
+            Vector3 rightHandPos = Vector3.right * 20.0f;
+            Vector3 leftHandPos = Vector3.right * -20.0f;
+
+            yield return null;
+
+            InputSimulationService inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+
+            // Test orbital around right hand
+            yield return TestHandSolver(testObjects.target, inputSimulationService, rightHandPos, Handedness.Right);
+
+            // Test orbital around left hand
+            yield return TestHandSolver(testObjects.target, inputSimulationService, leftHandPos, Handedness.Left);
+        }
+
+        /// <summary>
+        /// Test Surface Magnetism against "wall" and that attached object falls head direction
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestSurfaceMagnetism()
+        {
+            // Reset view to origin
+            MixedRealityPlayspace.PerformTransformation(p =>
+            {
+                p.position = Vector3.zero;
+                p.LookAt(Vector3.forward);
+            });
+
+            // Build wall to collide against
+            var wall = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            wall.transform.localScale = new Vector3(25.0f, 25.0f, 0.2f);
+            wall.transform.position = Vector3.forward * 10.0f;
+
+            yield return null;
+
+            // Instantiate our test gameobject with solver. 
+            // Set layer to ignore raycast so solver doesn't raycast itself (i.e BoxCollider)
+            var testObjects = InstantiateTestSolver<SurfaceMagnetism>();
+            testObjects.target.layer = LayerMask.NameToLayer("Ignore Raycast");
+
+            yield return new WaitForSeconds(SolverUpdateWaitTime);
```

Comment:
>WaitForSeconds(SolverUpdateWaitTime) [](start = 29, length = 36)

Does this actually add 2 real seconds of wait per test? Or is this "game time" which can be configured to move faster than normal?

Reply:
@wiwei it is real time. We can & should totally use game time though. Although I think the solvers are being updated against realtimeSinceStartup

Reply:
@wiwei  yea problem is TimeScale ( https://docs.unity3d.com/ScriptReference/Time-timeScale.html ) does not affect realtimesincestartup which is used by the solvers